src/votekit/models.py

```diff
@@ -1,95 +1,292 @@
+            that converts profiles to a score dictionary mapping candidates to their current score.
+            Used in creating ElectionState objects. Defaults to None.
+        length (int): the number of rounds of the election.
     """
 
-    def __init__(self, profile: PreferenceProfile, ballot_ties: bool = True):
-        if ballot_ties:
-            self._profile = self.resolve_input_ties(profile)
-        else:
-            self._profile = profile
+    def __init__(
+        self,
+        profile: PreferenceProfile,
+        score_function: Optional[
+            Callable[[PreferenceProfile], dict[str, float]]
+        ] = None,
+        sort_high_low: bool = True,
+    ):
+        self._profile = profile
+        self.score_function = score_function
+        self.sort_high_low = sort_high_low
+        self.election_states: list[ElectionState] = []
+        self._run_election()
+        self.length = len(self.election_states) - 1
+
+    def get_profile(self, round_number: int = -1) -> PreferenceProfile:
+        """
+        Fetch the PreferenceProfile of the given round number.
+
+        Args:
+            round_number (int, optional): The round number. Defaults to -1,
+                which accesses the final profile.
+
+        Returns:
+            PreferenceProfile
+
+        """
+        if round_number < -1:
+            raise ValueError("round_number must be -1 or non-negative.")
+
+        if round_number == -1:
+            round_number = len(self)
+
+        profile = self._profile
+
+        for i in range(round_number):
+            profile = self._run_step(profile, self.election_states[i])
+
+        return profile
+
+    def get_step(
+        self, round_number: int = -1
+    ) -> tuple[PreferenceProfile, ElectionState]:
+        """
+        Fetches the profile and ElectionState of the given round number.
+
+        Args:
+            round_number (int, optional): The round number to fetch. Defaults to -1,
+                which fetches the final round.
+
+        Returns:
+            tuple[PreferenceProfile, ElectionState]
+        """
+        if round_number < -1:
+            raise ValueError("round_number must be -1 or non-negative.")
+
+        if round_number == -1:
+            round_number = len(self)
+
+        return (self.get_profile(round_number), self.election_states[round_number])
+
+    def get_elected(self, round_number: int = -1) -> list[frozenset[str]]:
+        """
+        Fetch the elected candidates up to the given round number.
+
+        Args:
+            round_number (int, optional): Round number, defaults to -1 which corresponds to
+                the final round.
+
+        Returns:
+            list[frozenset[str]]: List of winning candidates in order of election. Candidates
+                in the same set were elected simultaneously, i.e. in the final ranking
+                they are tied.
+        """
+        if round_number < -1:
```

Comment:
You don't need this value error since the list type will take care of erroneous indexing for you